File: my_solution.js
function fibonacci(n) {
    let fib = [0, 1]
    
    if(n <= 1) return fib[n]
    
    for(let i = 2; i <= n; i++){
        fib.push(fib[i-1] + fib[i-2])
    }
    console.log('fib', fib)
    return fib.pop()
}

// Test Cases
console.log(fibonacci(0)) // 0
console.log(fibonacci(1)) // 1
console.log(fibonacci(10)) // 55
console.log(fibonacci(20)) // 6765

File: my_solution.py
def fibonacci(n: int) -> int:
    fib = [0, 1]
    if n <= 1:
        return fib[n]
    for i in range(2, n + 1):
        fib.append(fib[i - 2] + fib[i - 1])
    return fib[-1]   

# Test Cases
print(fibonacci(0)) # 0
print(fibonacci(1)) # 1
print(fibonacci(10)) # 55
print(fibonacci(20)) # 6765
